src/wp-includes/class-wp-query.php

```diff
@@ -3614,6 +3614,12 @@ public function get_queried_object() {
 			$this->queried_object_id = (int) $this->post->ID;
 		} elseif ( $this->is_author ) {
 			$this->queried_object_id = (int) $this->get( 'author' );
+			if ( ! $this->queried_object_id ) {
+				$user = get_user_by( 'slug', $this->get( 'author_name' ) );
```

Comment:
In your workaround PR you first checked whether `$this->get( 'author_name' )` is truthy. Worth doing that here as well?

Reply:
Good point, I've updated the code in the commit to match how it's done for taxonomies.